### PR TITLE
Adds an Apache Guacamole guide

### PR DESCRIPTION
**Pull request by ZPain8464**
This PR adds a guide to secure access to an Apache Guacamole instance.

### CODE REVIEW

File: content/examples/guacamole/guacamole.docker-compose.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```yaml title="docker-compose.yaml"
+
```
Comment (gaurdro) [inline content/examples/guacamole/guacamole.docker-compose.md:2]: Is this file meant to be this empty?

Comment (ZPain8464) [reply content/examples/guacamole/guacamole.docker-compose.md:2]: I forgot about that. I was going to move the examples into that directory, but we haven't been consistent about it. We can remove this.

**Comment by gaurdro** [comment]
Screenshot 2024-08-06 at 3 39 35 PM
we might want to drop the `Secure` from the front of that link to match the rest.

**Comment by ZPain8464** [comment]
@gaurdro I actually just updated that, but thanks for catching it.

File: content/docs/guides/guacamole.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+# cSpell:ignore guacd, GUACD, guacnetwork
+title: Secure Apache Guacamole
+description: This guide shows you to secure Apache Guacamole behind Pomerium using JWT Claims Headers to support HTTP header authentication.
+keywords: [JWT claims headers, http auth, header auth]
+sidebar_label: Apache Guacamole
+---
+
+# Secure Apache Guacamole
+
+This guide shows you how to secure an instance of [Apache Guacamole](https://guacamole.apache.org/), a clientless, remote desktop gateway that supports VNC, RDP, and SSH protocols.
```
Comment (desimone) [inline content/docs/guides/guacamole.md:11]: Lets clarify the objective at the start &  explicitly state the goal in the introduction, mentioning that the guide will help users set up a secure, authenticated Apache Guacamole instance behind Pomerium.

Comment (desimone) [inline content/docs/guides/guacamole.md:11]:
```suggestion
This guide will walk you through securing an instance of [Apache Guacamole](https://guacamole.apache.org/) behind Pomerium, leveraging JWT Claims Headers for HTTP header authentication. The setup allows centralized, secure, and seamless authentication via Pomerium. [Apache Guacamole](https://guacamole.apache.org/), a clientless, remote desktop gateway that supports VNC, RDP, and SSH protocols.
```

File: content/docs/guides/guacamole.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
```
Comment (desimone) [inline content/docs/guides/guacamole.md:12]:
```suggestion
+To complete this guide, you will:
+1. Install and configure Guacamole using Docker Compose.
+2. Replace the default Nginx with Pomerium in the Docker setup.
+3. Configure Pomerium to forward identity headers to Guacamole.
+4. Securely run your setup and verify proper authentication.
```

File: content/docs/guides/guacamole.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+:::
+
+## Install Guacamole with Docker Compose
+
+[Installing Guacamole](https://guacamole.apache.org/doc/gug/guacamole-docker.html) with Docker requires the following services:
+
+- `guacamole/guacd`: The guacd daemon, which provides support for VNC, RDP, SSH, telnet, and Kubernetes.
+- `guacamole/guacamole`: The Guacamole web application running within a Tomcat 8 server with websocket support.
+- `mysql` or `postgres`: The relational database that stores authentication and connection configuration data.
+
```
Comment (desimone) [inline content/docs/guides/guacamole.md:45]:
```suggestion
> **Security Best Practice:** Ensure that environment variables containing sensitive data, such as passwords, are stored securely. Consider using Docker secrets or an `.env` file.

```

File: content/docs/guides/guacamole.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+         - allow:
+             or:
+               - email:
+                   # highlight-next-line
+                   is: user@example.com
+       pass_identity_headers: true
+   ```
+
+   Don't forget to replace `user@example.com` with your own email address.
+
```
Comment (desimone) [inline content/docs/guides/guacamole.md:150]: Can you show the full working configurations / yamls at the end?

**Review by desimone — COMMENTED** [review]
Read for content but going to try and install on my homelab this weekend :)

File: content/docs/guides/guacamole.md (hunk 5 above)
Comment (ZPain8464) [reply content/docs/guides/guacamole.md:155]: Yes, I'll add a section to do that.

**Comment by desimone** [comment]
btw -- suggestions should be taken with giant grain of thought and probably need to be re-integrated / cleaned up. I was just shooting from the hip.

**Comment by ZPain8464** [comment]
@desimone thanks, I understand. I committed your suggestions and will work through them as I make further changes.

File: content/docs/guides/guacamole.md (hunk 2 above)
Comment (ZPain8464) [reply content/docs/guides/guacamole.md]: I rewrote the intro based on your feedback.

**Comment by desimone** [comment]
Could we make the sidebar items alphabetical

**Comment by ZPain8464** [comment]
@desimone yes, will do. Since it's a small ask I'll do that in this PR.

**Review by desimone — APPROVED** [review]
Can't wait to run through this myself. Copy LGTM.

**Pull request merged**